sh.c

```diff
@@ -17,10 +27,28 @@ FILE* create_shell_script(char *tmp_path_buffer) {
     time_t rawtime;
     time (&rawtime);
     fprintf(f, "# This shell script is generated by overlayfs-tools on %s\n", ctime (&rawtime));
+    for (int i=0; i < NUM_VARS; i++)
+        if (vars[i])
+            fprintf(f, "%s='%s'\n", var_names[i], vars[i]);
```

Comment:
It may not work well when `vars[i]` contains single quotes.

Reply:
in main.c vars[] is assigned a string returned from realpath(3). It does not contain any quotes, even if quotes are passed in command line

Reply:
I mean the case where the actual directory name itself include quotes.

Reply:
Can you show this with an example?
Actually run overlay that generates a bad script?

Reply:
Here's an example: I set up it with `mkdir "/tmp/test's upper"; mkdir "/tmp/test's lower"; touch "/tmp/test's upper/foo"`, and run `sudo ./overlay merge -l /tmp/test\'s\ lower/ -u /tmp/test\'s\ upper/`.

The generated shell script looks like:

```
LOWERDIR='/tmp/test's lower'
UPPERDIR='/tmp/test's upper'
rm -rf $LOWERDIR'/foo.txt'
```

The the single quote in `test's` breaks the string literal, and the script fails to execute.

Reply:
I see. I am not an expect in quoting as you can see.
So I need to fix create_shell_script() to use quote().
Let me try to fix that.

Reply:
done. please verify I did not miss anything